src/scripts/ci/gh_clang_tidy_fixes_in_pr.py

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env python3
+
+"""
+(C) 2023 Jack Lloyd
+    2023 René Meusel - Rohde & Schwarz Cybersecurity
+
+Botan is released under the Simplified BSD License (see license.txt)
+"""
+
+import argparse
+import glob
+import os
+import sys
+
+import yaml
+
+class FileLocation:
+    def __init__(self, line : int, column : int, endline : int | None = None, endcolumn : int | None = None):
+        self.line = line
+        self.column = column
+        self.endline = endline
+        self.endcolumn = endcolumn
+
+    def has_range(self):
+        return self.endline is not None and self.endcolumn is not None
+
+
+class Diagnostic:
+    @staticmethod
+    def read_diagnostics_from(yaml_file : str):
+        with open(yaml_file, encoding="utf-8") as yml:
+            fixes = yaml.load(yml, Loader=yaml.FullLoader)
+            if "Diagnostics" not in fixes:
+                raise RuntimeError(f"No Diagnostics found in {yaml_file}")
+            return [Diagnostic(diag) for diag in fixes["Diagnostics"]]
+
+
+    def __map_file_path(self, file_path, base_path): # pylint: disable=unused-argument
+        if file_path.endswith(".h"):
+            raise RuntimeError(f"Header file {file_path} cannot be handled")
+        # TODO: try to map include files (residing in build/include) onto their
+        #       origin path in src/lib etc.
```

Comment:
This might become urgent once https://github.com/randombit/botan/issues/3644 is done.